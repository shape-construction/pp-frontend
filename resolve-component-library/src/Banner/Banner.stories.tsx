import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { LockClosedIcon } from '../Icons/solid';
import { Banner } from '..';

export default {
  title: 'Banner',
  component: Banner,
} as ComponentMeta<typeof Banner>;

const Template: ComponentStory<typeof Banner> = (args) => <Banner {...args} />;

export const Default = Template.bind({});
Default.args = {
  message: 'This is a banner message',
  icon: <LockClosedIcon />,
};

export const Floating = Template.bind({});
Floating.args = {
  ...Default.args,
  floating: true,
};

export const Centered = Template.bind({});
Centered.args = {
  ...Default.args,
  centered: true,
};
