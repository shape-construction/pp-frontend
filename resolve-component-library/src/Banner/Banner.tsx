import classNames from 'classnames';
import React, { ReactNode } from 'react';

export type BannerProps = {
  /**
   * Icon to render before the children
   */
  icon: ReactNode;
  /**
   * Message to display in banner
   */
  message: string;
  /**
   * Floating variant
   */
  floating?: boolean;
  /**
   * Centered variant
   */
  centered?: boolean;
};

export const Banner: React.FC<BannerProps> = ({ icon, message, floating, centered }) => (
  <div
    className={classNames('flex w-full items-center gap-x-2.5 bg-yellow-50', {
      'rounded-md border border-yellow-100 p-3': floating,
      'p-1.5': !floating,
      'justify-center': centered,
    })}
    role="alert"
  >
    <span className="h-5 w-5 text-yellow-400">{icon}</span>
    <span className="text-sm font-medium text-yellow-800">{message}</span>
  </div>
);
