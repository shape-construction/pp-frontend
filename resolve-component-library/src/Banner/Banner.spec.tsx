import { render, screen } from '@testing-library/react';
import React from 'react';
import { Banner, BannerProps } from './Banner';

describe('Banner', () => {
  it('renders the message', () => {
    const props: BannerProps = {
      message: 'This is a banner message',
      icon: <div>a icon</div>,
    };
    render(<Banner {...props} />);

    expect(screen.getByText('This is a banner message')).toBeInTheDocument();
  });

  it('renders the icon', () => {
    const props: BannerProps = {
      message: 'This is a banner message',
      icon: <div>a icon</div>,
    };
    render(<Banner {...props} />);

    expect(screen.getByText('a icon')).toBeInTheDocument();
  });

  it('renders as floating', () => {
    const props: BannerProps = {
      message: 'This is a banner message',
      icon: <div>a icon</div>,
      floating: true,
    };

    const { container } = render(<Banner {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('renders as centered', () => {
    const props: BannerProps = {
      message: 'This is a banner message',
      icon: <div>a icon</div>,
      centered: true,
    };

    const { container } = render(<Banner {...props} />);

    expect(container).toMatchSnapshot();
  });
});
