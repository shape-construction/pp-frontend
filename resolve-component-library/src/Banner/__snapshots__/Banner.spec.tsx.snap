// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders as centered 1`] = `
<div>
  <div
    class="flex w-full items-center gap-x-2.5 bg-yellow-50 p-1.5 justify-center"
    role="alert"
  >
    <span
      class="h-5 w-5 text-yellow-400"
    >
      <div>
        a icon
      </div>
    </span>
    <span
      class="text-sm font-medium text-yellow-800"
    >
      This is a banner message
    </span>
  </div>
</div>
`;

exports[`Banner renders as floating 1`] = `
<div>
  <div
    class="flex w-full items-center gap-x-2.5 bg-yellow-50 rounded-md border border-yellow-100 p-3"
    role="alert"
  >
    <span
      class="h-5 w-5 text-yellow-400"
    >
      <div>
        a icon
      </div>
    </span>
    <span
      class="text-sm font-medium text-yellow-800"
    >
      This is a banner message
    </span>
  </div>
</div>
`;
